e2e/consulcompat/shared_setup_test.go

```diff
@@ -0,0 +1,117 @@
+// Copyright (c) HashiCorp, Inc.
+// SPDX-License-Identifier: BUSL-1.1
+
+package consulcompat
+
+import (
+	"io"
+	"os"
+	"path/filepath"
+	"testing"
+
+	consulapi "github.com/hashicorp/consul/api"
+	consulTestUtil "github.com/hashicorp/consul/sdk/testutil"
+	nomadapi "github.com/hashicorp/nomad/api"
+	"github.com/hashicorp/nomad/helper/testlog"
+	"github.com/hashicorp/nomad/helper/uuid"
+	"github.com/hashicorp/nomad/testutil"
+	"github.com/shoenig/test/must"
+)
+
+const (
+	consulDataDir = "consul-data"
+)
+
+// startConsul runs a Consul agent with bootstrapped ACLs and returns a stop
+// function, the HTTP address, and a HTTP API client
+func startConsul(t *testing.T, b build, baseDir, ns string) (string, *consulapi.Client) {
+
+	path := filepath.Join(baseDir, binDir, b.Version)
+	cwd, _ := os.Getwd()
+	os.Chdir(path)      // so that we can launch Consul from the current directory
+	defer os.Chdir(cwd) // return to the test dir so we can find job files
+
+	oldpath := os.Getenv("PATH")
+	os.Setenv("PATH", path+":"+oldpath)
+	t.Cleanup(func() {
+		os.Setenv("PATH", oldpath)
+	})
```

Comment:
I know already merged (good!), but fyi `testing.T` has a nice lil convenience method for this, which also prevents accidentally running it in parallel: [`t.Setenv`](https://pkg.go.dev/testing#T.Setenv)

```suggestion
	oldpath := os.Getenv("PATH")
	t.Setenv("PATH", path+":"+oldpath)
```

Reply:
TIL! I'm almost certainly going to touch these files again after the ENT test gets written. I'll pick that up then.